libraries/libfc/include/fc/crypto/sha3.hpp

```diff
@@ -56,6 +65,13 @@ class sha3
 		fc::fwd<impl, 1016> my;
 	};
 
+	struct nist_encoder : public encoder {
+		sha3 result() { return encoder::result(true); }
+	};
+	struct keccak_encoder : public encoder {
+		sha3 result() { return encoder::result(false); }
+	};
+
```

Comment:
sha3 is funky because of this nist vs keccak stuff going on. I opted to keep with the tag dispatching but it means more bespoke plumbing for this hash type & packhash()